ProductsWebApi/ProductsWebApi/Repository/UserRepository.cs

```diff
@@ -0,0 +1,25 @@
+﻿using ProductsWebApi.DBContext;
+using ProductsWebApi.Interface;
+using ProductsWebApi.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace ProductsWebApi.Repository
+{
+    public class UserRepository : IUserRepository
+    {
+        private readonly IProductsDBContext _productsDBContext;
+
+        public UserRepository(IProductsDBContext productsDBContext)
+        {
+            _productsDBContext = productsDBContext;
+        }
+
+        public async Task<bool> FetchUserDetails(string userName, string password)
+        {
+            return _productsDBContext.Users.Any(user => user.UserName == userName && user.Password == password);
```

Comment:
not sure, why this hasn't been awaited? Is the functionality works?? still? 
Prefer await for maintaining asynchronous operations

Reply:
made the operation asynchronous and used await